geode-core/src/main/java/org/apache/geode/cache/client/internal/ExecuteRegionFunctionOp.java

```diff
@@ -641,10 +641,22 @@ protected Object processResponse(Message msg) throws Exception {
     }
 
     private void addFunctionException(final FunctionException result) {
-      if (this.functionException == null) {
-        this.functionException = result;
+      if (result instanceof FunctionInvocationTargetException) {
+        if (this.functionException == null) {
+          this.functionException = new FunctionException(result);
+        }
+        this.functionException.addException(result);
```

Comment:
Isn't this addException logic the same in all three conditional branches?  Seems like this could be extracted out of the conditional and called once at the bottom of this method.

Reply:
Nevermind, I now see that this.functionException changes in each branch.